tests/handlers/test_sliding_sync.py

```diff
@@ -3213,3 +3216,402 @@ def test_default_bump_event_types(self) -> None:
+        request_required_state_map = {"type1": {"state_key"}}
+
+        to_persist, added = _required_state_changes(
+            user_id="@user:test",
+            previous_room_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=previous_required_state_map,
+            ),
+            room_sync_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=request_required_state_map,
+            ),
+            state_deltas={},
+        )
+
+        # We've removed a type without a change, so nothing should change.
+        self.assertIsNone(to_persist)
+        self.assertEqual(added, StateFilter.none())
+
+    def test_simple_remove_type_change(self) -> None:
+        """Test removing a type from the config when there are is a matching
+        state delta does cause the persisted required state config to change"""
+
+        previous_required_state_map = {"type1": {"state_key"}, "type2": {"state_key"}}
+        request_required_state_map = {"type1": {"state_key"}}
+
+        to_persist, added = _required_state_changes(
+            user_id="@user:test",
+            previous_room_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=previous_required_state_map,
+            ),
+            room_sync_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=request_required_state_map,
+            ),
+            state_deltas={("type2", "state_key"): "$event_id"},
+        )
+
+        # We've removed a type and there's been a change to that state, so we
+        # persist the change to required state.
+        assert to_persist is not None
+        self.assertDictEqual(to_persist, request_required_state_map)
+        self.assertEqual(added, StateFilter.none())
+
+    def test_type_wildcards_add(self) -> None:
+        """Test that changing the wildcard type"""
+
+        previous_required_state_map = {"type1": {"state_key"}}
+        request_required_state_map = {"type1": {"state_key"}, "*": {"state_key"}}
+
+        to_persist, added = _required_state_changes(
+            user_id="@user:test",
+            previous_room_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=previous_required_state_map,
+            ),
+            room_sync_config=RoomSyncConfig(
+                timeline_limit=0,
+                required_state_map=request_required_state_map,
+            ),
+            state_deltas={},
+        )
+
+        # We've added a wildcard, so we persist the change and request everything
+        assert to_persist is not None
+        self.assertDictEqual(to_persist, request_required_state_map)
+        self.assertEqual(added, StateFilter.all())
+
+    def test_type_wildcards_remove(self) -> None:
+        """Test that changing the wildcard type"""
+
+        previous_required_state_map = {"type1": {"state_key"}, "*": {"state_key"}}
```

Comment:
Prefer the constants so we can easily find the test cases when looking at the code for `StateValues.WILDCARD` things.

```suggestion
        previous_required_state_map = {"type1": {"state_key"}, StateValues.WILDCARD: {"state_key"}}
```

Reply:
Updated in https://github.com/element-hq/synapse/pull/17805